### ISSUE TITLE
Menu does not close when switching between pages

### ISSUE THREAD

**Issue by smiera** [salim-lachdhaf/searchable_dropdown#554]
Here is a simple example: https://gist.github.com/smiera/762c200059a2960f142bc44466ecbdf9

1. On page `Menu #1` open dropdown
2. Switch to page `Menu #2`. Opened dropdown list is still open.

If I remove NoTransitionPage in GoRouter(replace `pageBuilder` to `builder` in example), then everything works correctly. 

![Simulator Screen Recording - iPhone 12 - 2023-03-21 at 10 08 05](https://user-images.githubusercontent.com/3142583/226548732-bc7b387d-38d9-441d-b68d-1a4ebb0628d7.gif)

```
dependencies:
  go_router: ^6.0.3
  dropdown_search: ^5.0.6
```

**Comment by tobiasleijs** [comment]
I have the same issue

**Comment by FlorisGC** [comment]
I have the same issue

**Comment by NgoTruongThanh** [comment]
I have the same issue

**Comment by salim-lachdhaf** [comment]
thank you for the example.

the issue is fixed in the next version v6.0.0-pre that will be released in the next days.

Stay tuned ;)
